Ronin.rte/Devices/Tools/Shovel/Shovel.lua

```diff
@@ -79,13 +80,36 @@ function Update(self)
 
 			self.Magazine.RoundCount = resource > 0 and resource or -1;
 		end
-		self.RateOfFire = self.minimumRoF + (self.minimumRoF) * (actor.Health/actor.MaxHealth);
+		self.RateOfFire = self.minimumRoF + (self.minimumRoF) * (parent.Health/parent.MaxHealth);
 	else
 		self.Scale = 1;
 		if self.Magazine then
 			self.Magazine.Scale = 0;
 		end
+		if self.lastVel.Magnitude > 25 then
+			if self.HitWhatMOID ~= rte.NoMOID then
+				local mo = MovableMan:GetMOFromID(self.HitWhatMOID);
+				if mo then
+					local particleCount = 3;
+					local spread = self.AngularVel * TimerMan.DeltaTimeSecs * 0.5;
+					for i = 0, particleCount - 1 do
+						local damagePar = CreateMOPixel("Smack Particle");
+						damagePar.Mass = self.Mass--/particleCount;
+						damagePar.Sharpness = self.Sharpness-- * particleCount;
+
+						damagePar.Pos = self.lastMuzzlePos;
+						damagePar.Vel = Vector(self.lastVel.X, self.lastVel.Y):RadRotate(spread * 0.5 - spread * i/(particleCount - 1)) * 1.5;
+	
+						damagePar:SetWhichMOToNotHit(self, -1);
+						MovableMan:AddParticle(damagePar);
+					end
+					self.hitSound:Play(self.MuzzlePos);
+				end
+			end
+		end
 	end
+	self.lastVel = Vector(self.Vel.X, self.Vel.Y);
```

Comment:
This actually already exists in cpp, MovableObject has a m_PrevVel field. You could expose it to lua if you wanted to, obviously readonly.